jdaviz/app.py

```diff
@@ -686,8 +687,13 @@ def get_data_from_viewer(self, viewer_reference, data_label=None,
                         if cls is not None:
                             handler, _ = data_translator.get_handler_for(cls)
                             try:
-                                layer_data = handler.to_object(layer_data,
-                                                               statistic=statistic)
+                                if cls == Spectrum1D:
```

Comment:
is there any other `cls` that we need to support for collapsed cubes in cubeviz?

Reply:
There's been talk of visualizing ramps in a Cubeviz-like helper (cc @camipacifici), but I don't think we need to worry too much about supporting that "correctly" yet?

Reply:
ok, but right now there is no other class that we support mapping (ndarray, ccddata, etc)?

Reply:
We often check against `cls` is `Spectrum1D` to check if a `Data` object is a spectrum. I think we'd have to do some work to generalize `app.py` to representing ND spectra with other classes.